stixcore/products/product.py

```diff
@@ -38,6 +42,18 @@ def __init_subclass__(cls, **kwargs):
         if hasattr(cls, 'is_datasource_for'):
             cls._registry[cls] = cls.is_datasource_for
 
+    @classmethod
+    def from_levelb(cls, levelb):
+
+        packets = [Packet(unhexlify(d)) for d in levelb.data['data']]
+
+        for packet in packets:
+            decompression.decompress(packet)
```

Comment:
This is actually a better approach as its possible that compression or calibration parameters could change between packets.